pkg/gluster-virtblock/controllerserver.go

```diff
@@ -43,11 +45,90 @@ type CsiDrvParam struct {
 	CsiDrvVersion    string
 }
 
+func setReplicaCnt(volumeReq *bapi.BlockVolumeCreateRequest, value string) error {
+	replicaCnt, err := utils.ParseVolumeParamInt("replicas", value)
+	if err != nil {
+		return err
+	}
+	volumeReq.HostVolumeInfo.HostVolReplicaCnt = replicaCnt
+	return nil
+}
+func setArbiterReplicaCnt(req *csi.CreateVolumeRequest, volumeReq *bapi.BlockVolumeCreateRequest, value string) error {
+	if value == "thin" {
+		if err := utils.ValidateThinArbiter(req); err != nil {
+			return err
+		}
+		volumeReq.HostVolumeInfo.HostVolReplicaCnt = 2
+	} else {
+		return fmt.Errorf("invalid arbiterType specified: %s", value)
+	}
+	return nil
+}
+
+func setArbiterPath(req *csi.CreateVolumeRequest, volumeReq *bapi.BlockVolumeCreateRequest, value string) error {
+	if _, ok := req.Parameters["arbiterType"]; !ok {
+		glog.Error("only arbiterPath provided, missing arbiterType")
+		return fmt.Errorf("only arbiterPath provided, missing arbiterType")
+	}
+	volumeReq.HostVolumeInfo.HostVolThinArbPath = value
+	return nil
+}
+
+func setShardSize(volumeReq *bapi.BlockVolumeCreateRequest, value string) error {
+	shardSize, err := strconv.ParseUint(value, 10, 64)
+	if err != nil {
+		return fmt.Errorf("shardSize value: %s must be an integer", value)
+	}
+	volumeReq.HostVolumeInfo.HostVolShardSize = shardSize
+	return nil
+}
+
+func setHostVolumeSize(volumeReq *bapi.BlockVolumeCreateRequest, value string) error {
+	hostVolumeSize, err := strconv.ParseUint(value, 10, 64)
+	if err != nil {
+		return fmt.Errorf("hostVolumeSize value: %s must be an integer", value)
+	}
+	volumeReq.HostVolumeInfo.HostVolSize = hostVolumeSize
+	return nil
+}
+
+func parseHostVolumeParams(req *csi.CreateVolumeRequest, volumeReq *bapi.BlockVolumeCreateRequest) error {
+	var err error
+	for k, v := range req.GetParameters() {
+		switch k {
+		case "replicas":
+			err = setReplicaCnt(volumeReq, v)
+		case "arbiterType":
+			err = setArbiterReplicaCnt(req, volumeReq, v)
+		case "arbiterPath":
+			err = setArbiterPath(req, volumeReq, v)
+		case "shardSize":
+			err = setShardSize(volumeReq, v)
+		case "hostVolumeSize":
```

Comment:
need to add this value to storage-class example and please  update readme what this value  meant to be